tool/tctl/common/acl_command.go

```diff
@@ -0,0 +1,242 @@
+}
+
+// TryRun takes the CLI command as an argument (like "acl ls") and executes it.
+func (c *ACLCommand) TryRun(ctx context.Context, cmd string, client auth.ClientI) (match bool, err error) {
+	switch cmd {
+	case c.list.FullCommand():
+		err = c.List(ctx, client)
+	case c.show.FullCommand():
+		err = c.Show(ctx, client)
+	case c.usersAdd.FullCommand():
+		err = c.UsersAdd(ctx, client)
+	case c.usersRemove.FullCommand():
+		err = c.UsersRemove(ctx, client)
+	case c.usersList.FullCommand():
+		err = c.UsersList(ctx, client)
+	default:
+		return false, nil
+	}
+	return true, trace.Wrap(err)
+}
+
+// List will list access lists visible to the user.
+func (c *ACLCommand) List(ctx context.Context, client auth.ClientI) error {
+	accessLists, err := client.AccessListClient().GetAccessLists(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if len(accessLists) == 0 {
+		fmt.Println("no access lists")
+		return nil
+	}
+
+	return trace.Wrap(displayAccessLists(c.format, accessLists...))
+}
+
+// Show will display information about an access list visible to the user.
+func (c *ACLCommand) Show(ctx context.Context, client auth.ClientI) error {
+	accessList, err := client.AccessListClient().GetAccessList(ctx, c.accessListName)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	return trace.Wrap(displayAccessLists(c.format, accessList))
+}
+
+// UsersAdd will add a user to an access list.
+func (c *ACLCommand) UsersAdd(ctx context.Context, client auth.ClientI) error {
+	expires, err := time.Parse(time.RFC3339, c.expires)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	accessList, err := client.AccessListClient().GetAccessList(ctx, c.accessListName)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	accessList.Spec.Members = append(accessList.Spec.Members, accesslist.Member{
+		Name:    c.userName,
+		Reason:  c.reason,
+		Expires: expires,
+
+		// The following fields will be updated in the backend, so their values here don't matter.
+		Joined:  time.Now(),
+		AddedBy: "dummy",
+	})
+
+	_, err = client.AccessListClient().UpsertAccessList(ctx, accessList)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	fmt.Printf("successfully added user %s to access list %s", c.userName, c.accessListName)
+
+	return nil
+}
+
+// UsersRemove will remove a user to an access list.
+func (c *ACLCommand) UsersRemove(ctx context.Context, client auth.ClientI) error {
+	accessList, err := client.AccessListClient().GetAccessList(ctx, c.accessListName)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	memberIndex := -1
+	for i, member := range accessList.Spec.Members {
+		if member.Name == c.userName {
+			memberIndex = i
+			break
+		}
+	}
+
+	if memberIndex == -1 {
+		return trace.NotFound("user %s is not a member of access list %s\n", c.userName, c.accessListName)
+	}
+
+	accessList.Spec.Members = append(accessList.Spec.Members[:memberIndex], accessList.Spec.Members[memberIndex+1:]...)
```

Comment:
`slices.Insert` would work here too

Reply:
Wouldn't you want `slices.Delete` instead?

Reply:
Lol! You've only made the argument stronger. Every time I see this slice magic I have to figure out what it's actually doing, and using a (soon to be standard) function with a descriptive name takes this cognitive burden away. :-)

Reply:
Slices is blowing my mind, didn't even know it existed. Will make the change!

Reply:
It's going to be in the standard library starting in Go 1.21 🙂